wandb/sdk/wandb_run.py

```diff
@@ -1669,9 +1669,28 @@ def log_artifact(self, artifact_or_path, name=None, type=None, aliases=None):
         if isinstance(aliases, str):
             aliases = [aliases]
         artifact.finalize()
+        self._assert_can_log_artifact(artifact)
         self._backend.interface.publish_artifact(self, artifact, aliases)
         return artifact
 
+    def _assert_can_log_artifact(self, artifact):
+        overrides = {"run": self.id}
+        run_obj = self._run_obj
```

Comment:
We have this logic duplicated now between `use_artifact` and here. Might be worth adding a `_public_api()` factory that handles this.

Reply:
nice call - added in next commit